Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -84,6 +91,248 @@ namespace AzFramework
+                }
+
+                if (HasMatchingDeviceAttributeRule(deviceRegistrar, registry, visitArgs.m_jsonKeyPath))
+                {
+                    // only one rule has to match
+                    hasMatchingRule = true;
+                    return AZ::SettingsRegistryInterface::VisitResponse::Done;
+                }
+
+                return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+            };
+
+            AZ::SettingsRegistryVisitorUtils::VisitObject(*registry, callback, jsonPath);
+
+            return hasMatchingRule;
+        }
+
+        struct RuleSetting
+        {
+            AZStd::string setting;
+            AZStd::string value;
+        };
+
+        // order matters with rules, the user may choose to use the first matching rule or the last (default)
+        // a list is used to maintain the order in which settings will be applied and a map is used to quickly
+        // check if settings are in the list and retrieve an iterator to the setting within the list
+        using RuleSettingList = AZStd::list<RuleSetting>;
+        using RuleIteratorMap = AZStd::unordered_map<AZStd::string, RuleSettingList::iterator>;
+
+        void UpdateRuleSettings(AZ::SettingsRegistryInterface* registry, RuleSettingList& settingsList, RuleIteratorMap& settingsMap, AZStd::string_view jsonPath, DeviceRuleResolution ruleResolution )
+        {
+            using Type = AZ::SettingsRegistryInterface::Type;
+            auto callback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+            {
+                // has another rule provided a value for this setting already?
+                auto iter = settingsMap.find(visitArgs.m_fieldName);
+                if (iter != settingsMap.end() && ruleResolution == DeviceRuleResolution::First)
+                {
+                    // when using "First" we ignore duplicate setting entries 
+                    return AZ::SettingsRegistryInterface::VisitResponse::Skip;
+                }
+
+                // convert the value to a string to pass to console for execution
+                AZStd::string stringValue;
+                switch (visitArgs.m_type)
+                {
+                case Type::FloatingPoint:
+                    if (double value; registry->Get(value, visitArgs.m_jsonKeyPath))
+                    {
+                        // handle Min/Max rule resolution for numbers
+                        if (iter != settingsMap.end() && ruleResolution != DeviceRuleResolution::Last)
+                        {
+                            double currentValue = 0;
+                            AZ::ConsoleTypeHelpers::StringToValue(currentValue, iter->second->value);
+                            if ((ruleResolution == DeviceRuleResolution::Min && currentValue <= value) ||
+                                (ruleResolution == DeviceRuleResolution::Max && currentValue >= value))

```

Comment:
Ok so the `currentValue` is the current Value that has been evaluated so far?
So `value` must be less than the current value in order to replace the value when Rule Resolution is a `Min` and greater than the current value in order to replace the value when Rule Resolution is `Max`

Can you add a comment about this function explaining it for other maintainers?